components/ifc-viewer.tsx

```
@@ -694,6 +735,272 @@ function ViewerContent() {
+      if (typeof data === 'number' || typeof data === 'boolean') return regexInstance.test(String(data).toLowerCase());
+
+      if (Array.isArray(data)) {
+        for (const item of data) {
+          if (recursiveSearch(item, regexInstance, searchKeys)) return true;
+        }
+        return false;
+      }
+
+      if (typeof data === 'object') {
+        // Handle IFC.js specific structures like { value: X } or { NominalValue: { value: Y } } etc.
+        if (data.hasOwnProperty('value')) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)